### ISSUE TITLE
Implement missing commonly used loss functions

### ISSUE THREAD

**Issue by vladimirjovanovicTT** [tenstorrent/tt-forge-fe#850]
Currently only two losses are implemented in FFE: L1 and CrossEntropy.

We need to support commonly used losses:
https://pytorch.org/docs/stable/generated/torch.nn.L1Loss.html#torch.nn.L1Loss

Some of the most important ones are (in order of importance):
- MSELoss
- NLLLoss
- KlDivLoss
- HuberLoss
- TripletMarginloss
- BCELoss
- BCEWithLogitsLoss
- CTC

To support training generality, we will want to match all framework losses at one point.
Please create individual issues for all of the above losses.
